compiler/noirc_evaluator/src/acir/acir_context/mod.rs

```diff
@@ -1169,6 +1169,18 @@ impl<F: AcirField, B: BlackBoxFunctionSolver<F>> AcirContext<F, B> {
         let r_is_not_0 = self.not_var(r_is_0, AcirType::unsigned(1))?;
         let remainder = self.mul_var(remainder, r_is_not_0)?;
 
+        // Issue #9847 - The quotient must be a valid signed integer.
```

Comment:
This is a very minor nit, but I think it would better if the issue was linked as part of the test instead, with a link to Github itself not just the issue number. I thought I left this comment with the original review but I must have accidentally deleted it.